packages/atomic/src/components/commerce/product-template-components/atomic-product-children/atomic-product-children.tsx

```diff
@@ -0,0 +1,188 @@
+import {Schema, StringValue} from '@coveo/bueno';
+import {Product, ChildProduct} from '@coveo/headless/commerce';
+import {
+  Component,
+  h,
+  Element,
+  Prop,
+  Event,
+  EventEmitter,
+  State,
+  Method,
+} from '@stencil/core';
+import {
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {Carousel} from '../../../common/carousel';
+import {CommerceBindings} from '../../atomic-commerce-interface/atomic-commerce-interface';
+import {ProductContext} from '../product-template-decorators';
+
+/**
+ * @internal
+ * The `atomic-product-children` component renders a section that allows the user to select a nested product (e.g., a color variant of a given product)
+ *
+ * This component leverages the [product grouping](https://docs.coveo.com/en/l78i2152/) feature.
+ */
+@Component({
+  tag: 'atomic-product-children',
+  styleUrl: 'atomic-product-children.pcss',
+  shadow: false,
+})
+export class AtomicProductChildren
+  implements InitializableComponent<CommerceBindings>
+{
+  @InitializeBindings() public bindings!: CommerceBindings;
+  @ProductContext() private product!: Product;
+
+  @Element() hostElement!: HTMLElement;
+
+  @State() private children: (Product | ChildProduct)[] = [];
+  @State() private currentPage = 0;
+
+  public error!: Error;
+  private originalParentProduct: Product | null = null;
+
+  // TODO: there's a totalNumberOfRelatedProducts property on the Product that could be used to do server-side pagination. However I'm not sure how to (or even if we can) fetch additional related products with the commerce API
+  // TODO: remove this prop; we'll want the carousel to automatically kick in at the proper responsive breakpoint instead.
+  @Prop() public childrenPerPage: number = 6;
+
+  /**
+   * The non-localized label to display for the product children section.
+   */
+  @Prop() public label: string = 'available-in';
+
+  @Event({
+    eventName: 'atomic/hoverChildProduct',
+    bubbles: true,
+  })
+  public hoverChildProduct!: EventEmitter<{
+    hoveredChildProduct: ChildProduct;
+    currentParentProductId: string;
+    originalParentProduct: Product;
+  }>;
+
+  constructor() {
+    this.validateProps();
+  }
+
+  public connectedCallback(): void {
+    this.originalParentProduct = this.isProduct(this.product.children[0])
```

Comment:
If we do the "swapProduct()" as part of Headless, I guess we could arrange for this logic to go away: The swap() could modify `children` so that it becomes a true `parent` (with an array of children)